File: frontend/lib/Model/news_model.dart
class NewsModel {
  String? title;
  String? description;
  String? imageUrl;
  String? postedByEmail;
  String? postedByName;
  DateTime? createdAt;

  NewsModel({
    required this.title,
    required this.description,
    required this.imageUrl,
    required this.postedByEmail,
    required this.postedByName,
    required this.createdAt,
  });

  factory NewsModel.fromJson(Map<String, dynamic> json) {
    return NewsModel(
      title: json['title'] as String,
      description: json['description'] as String,
      imageUrl: json['imageUrl'] as String,
      postedByEmail: json['postedByEmail'] as String,
      postedByName: json['postedByName'] as String,
      createdAt: json['createdAt'] as DateTime,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "title": title,
      "description": description,
      "imageUrl": imageUrl,
      "postedByEmail": postedByEmail,
      "postedByName": postedByName,
      "createdAt": createdAt,
    };
  }
}

File: backend/src/Model/news_model.js
const mongoose = require("mongoose");

const News = mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  imageUrl: {
    type: String,
    required: true,
  },
  postedByEmail: {
    type: String,
    required: true,
  },
  postedByName: {
    type: String,
    required: true,
  },
  createdAt: {
    type: Date,
    default: new Date(),
  },
});

module.exports = mongoose.model("News", News);
